internal/services/dataprotection/data_protection_backup_vault_resource.go

```diff
@@ -92,6 +100,16 @@ func resourceDataProtectionBackupVault() *pluginsdk.Resource {
 				ValidateFunc: validation.StringInSlice(backupvaults.PossibleValuesForSoftDeleteState(), false),
 			},
 
+			// NOTE O+C: we do not want to disable immutability if a user has enabled it via the portal or other methods.
+			"immutability": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				Computed:     true,
```

Comment:
I don't think we should have this as Computed just for people managing this outside of Terraform. It shouldn't be a breaking change for people not solely using this resource within Terraform